hilti/include/rt/types/network.h

```diff
@@ -21,20 +21,36 @@ class Network {
     /**
      * Constructs a network from prefix address and length.
      *
-     * @param prefix address, which's *length* lower bits will be masked out.
-     * @param prefix length, which must be in the range from 0-32 for IPv4
-     * addresses; and 0-128 for IPv6 addresses.
+     * @param prefix prefix address, which's *length* lower bits will be masked out.
+     * @param length prefix length, which must be in the range from 0-32 for IPv4
+     *        addresses; and 0-128 for IPv6 addresses.
+     *
+     * @throws InvalidArgument for invalid length values.
      */
-    Network(const Address& prefix, int length) : _prefix(prefix), _length(length) { _mask(); }
+    Network(const Address& prefix, int length) : _prefix(prefix), _length(length) {
+        switch ( _prefix.family() ) {
+            case AddressFamily::IPv4:
+                if ( _length < 0 || _length > 32 )
+                    throw InvalidArgument(fmt("prefix length %s is invalid for IPv4 networks", _length));
+                break;
+            case AddressFamily::IPv6:
+                if ( _length < 0 || _length > 128 )
+                    throw InvalidArgument(fmt("prefix length %s is invalid for IPv6 networks", _length));
+                break;
+            case AddressFamily::Undef: cannot_be_reached();
```

Comment:
Let's throw InvalidArgument here as we. `cannot_be_reached()` is meant for places that really logically cannot happen, but here could be a way to create an `Address` with an `Undef` family somehow.